OlinkAnalyze/R/olink_normalization_product.R

```diff
@@ -640,3 +640,80 @@ olink_normalization_qs <- function(lst_df,
 
   return(df_qq_norm)
 }
+
+
+
+#' Formatting the output of olink_normalization_product for seamless use with
+#' downstream OA functions.
+#'
+#' @author
+#' Danai G. Topouza
+#'
+#' @description
+#' Removes non-bridgeable assays. Replaces the NPX values of the non-reference
+#' project by the Median Centered or QS Normalized NPX, according to the
+#' Bridging Recommendation. Replaces OlinkID by the concatenation of the
+#' Explore HT and Explore 3072 OlinkIDs to record the OlinkIDs from both
+#' projects. Replaces SampleID with the concatenation of SampleID and Project
+#' to make unique sample IDs for downstream analysis. Removes internal and
+#' external controls. Removes BridgingRecommendation, MedianCenteredNPX,
+#'QSNormalizedNPX, OlinkID_E3072 columns.
+#'
+#' @param df A "tibble" of Olink data in long format resulting from the
+#' olink_normalization_product function.
+#'
+#' @return A "tibble" of Olink data in long format containing both input
+#' datasets with the quantile normalized quantifications, with the above
+#' modifications.
+#'
+#' @examples
+#' \donttest{
+#' # Bridge samples
+#' bridge_samples <- intersect(
+#'   x = unique(OlinkAnalyze:::data_ht_small$SampleID),
+#'   y = unique(OlinkAnalyze:::data_3k_small$SampleID)
+#' ) |>
+#'   (\(x) x[!grepl("CONTROL", x)])()
+#'
+#' # Run olink_normalization_product
+#' npx_br_data <- olink_normalization(
+#' df1 = OlinkAnalyze:::data_ht_small,
+#' df2 = OlinkAnalyze:::data_3k_small,
+#' overlapping_samples_df1 = bridge_samples,
+#' df1_project_nr = "Explore HT",
+#' df2_project_nr = "Explore 3072",
+#' reference_project = "Explore HT")
+#'
+#' # Format output
+#' npx_br_data_format <- OlinkAnalyze:::olink_normalization_product_format(
+#' npx_br_data)
+#'
+#' }
+
+olink_normalization_product_format <- function(df) {
+
+  ### Keep the data following BridgingRecommendation
+  df_format <- df |>
+    dplyr::filter(.data[["SampleType"]] == "SAMPLE") |> # Remove controls
+    dplyr::mutate(SampleID = paste0(.data[["SampleID"]],
+                                     "_",
+                                     .data[["Project"]])) |>
+    dplyr::filter(.data[["BridgingRecommendation"]] != "NotBridgeable") |>
```

Comment:
Removes NotBridgeable datapoints as expected per PR notes.

Question: do we want to remove these? Or can we keep the NPX as is in these cases and keep these lines in?

Reply:
The NotBridgeable assays don't have a recommendation for MedianCenteredNPX or QSNormalizedNPX, I'm not sure how to include them.

Reply:
Kathy suggested to keep them with the 3k NPX values and 3k OlinkID and the HT ones with just the HT OlinkID.

Reply:
I could not get this to work - downstream tests like t.test would throw an error, since the not bridgeable assay HT and 3072 OlinkIDs are only present in some samples and not others.